src/analysis/convex/cone/proper.lean

```diff
@@ -46,8 +47,76 @@ protected def closure (K : convex_cone ℝ E) : convex_cone ℝ E :=
 
 @[simp, norm_cast] lemma coe_closure (K : convex_cone ℝ E) : (K.closure : set E) = closure K := rfl
 
-protected lemma mem_closure {K : convex_cone ℝ E} {a : E} :
+@[simp] protected lemma mem_closure {K : convex_cone ℝ E} {a : E} :
   a ∈ K.closure ↔ a ∈ closure (K : set E) :=
 iff.rfl
 
 end convex_cone
+
+section definitions
+
+/-- A proper cone is a convex cone `K` that is nonempty and closed. Proper cones have the nice
+property that the dual of the dual of a proper cone is itself. This makes them useful for defining
+cone programs and proving duality theorems. -/
+structure proper_cone (E : Type*) [normed_add_comm_group E] [inner_product_space ℝ E]
+  [complete_space E] extends convex_cone ℝ E :=
```

Comment:
```suggestion
  extends convex_cone ℝ E :=
```

Reply:
I need the complete space structure to say that dual is involutive. This fact requires docs#convex_cone.inner_dual_cone_of_inner_dual_cone_eq_self